### DISCUSSION TITLE
Choosing Between Trivy and tfsec for Terraform Code Scanning

### DISCUSSION THREAD

**Discussion by faizu6** [aquasecurity/trivy#5165] [Q&A] [answered]
### Question

I recently scanned my private Terraform repository using both Trivy and tfsec. I noticed that Trivy's output contained almost twice as many findings as tfsec, which made me initially lean towards using Trivy. However, upon reviewing the documentation, I discovered that Trivy doesn't scan .tfvars files (as mentioned in version 0.19 [documentation](https://aquasecurity.github.io/trivy/v0.19.2/misconfiguration/comparison/tfsec/#:~:text=tfsec%20uses%20static%20analysis%20of,differences%20between%20Trivy%20and%20tfsec.)).
I'm aware that the current version is 0.45, and I'd like to know if Trivy has since included .tfvars file scanning. Additionally, I'm looking for guidance on whether I should choose Trivy over tfsec for Terraform code scanning, and what the key differences are between the current versions of Trivy and tfsec when scanning Terraform code. It's worth mentioning that I already use Trivy for vulnerability scanning in my workflow.

### Target

None

### Scanner

None

### Output Format

None

### Mode

None

### Version

```bash
Faizan:~$ trivy --version
Version: 0.45.0
Vulnerability DB:
  Version: 2
  UpdatedAt: 2023-09-11 00:17:18.649275472 +0000 UTC
  NextUpdate: 2023-09-11 06:17:18.649274372 +0000 UTC
  DownloadedAt: 2023-09-11 05:02:58.685858103 +0000 UTC
Policy Bundle:
  Digest: sha256:2e95a2d5d45de8ebecae53a97403230a6c608a579b082f3de170f3cf09e46243
  DownloadedAt: 2023-08-11 10:29:36.936906682 +0000 UTC
```

**Comment by nikpivkin** [comment, marked as answer]
Hi @faizu6 !

We've moved our efforts to Trivy https://github.com/aquasecurity/tfsec/discussions/1994.
New rules have been added to the Trivy and some bugs have been fixed, so the output may contain more results. You [can pass](https://aquasecurity.github.io/trivy/v0.45/tutorials/misconfiguration/terraform/#passing-tftfvars-files-into-trivy-config-scans) `.tfvars` when scanning configs.

**Reply by faizu6** [reply]
Thanks for the quick reply @nikpivkin  
Is it necessary to pass the `.tfvars` as config every time i scan a terraform code If I want to overide the default values mentioned in the Terraform HCL file.
Also can i handle all my exceptions using `.trivyignore` like for e.g if i dont want the output to show [ID AVD-AWS-0164](https://avd.aquasec.com/misconfig/aws/ec2/avd-aws-0164/#undefined) in the list of vulnerabilities found, I will add this to `.trivyignore` and it will never come up in the results.

**Reply by nikpivkin** [reply]
Yes, `.tfvars` needs to pass every time. You can specify the rule ID in `.trivyignore`, which Trivy should ignore. You can learn more about filtering the results in the [documentation](https://aquasecurity.github.io/trivy/v0.45/docs/configuration/filtering/).

**Comment by amit2067** [comment]
Hi, Could you please let me know how many rules does Trivy have, for the following? 

- Terraform
- Kubernetes
- Helm
- Docker

Is there an open source repository where I can find these rules and numbers.

**Reply by itaysk** [reply]
https://github.com/aquasecurity/trivy-policies/tree/275c474642c9ce66131245ae4134681c8b6f63af/rules
